test/hack/bin/verify-rules.sh

```diff
@@ -1,16 +1,23 @@
 #!/bin/bash
-set -euo pipefail
 #
 # This test ensures that all prometheus rules are valid
 #
 
-set -eu
+set -euo pipefail
 
 # Global options
 
 ## GENERATE_ONLY: if set to true, the script will only generate the rules files
 GENERATE_ONLY="${GENERATE_ONLY:-false}"
 
+## RULES_TYPES: list of supported rule types to test and their associated suffixes
+declare -A RULES_TYPES
+RULES_TYPES["prometheus"]="rules.yml"
+RULES_TYPES["loki"]="logs.yml"
+
+## RULES_TYPE_DEFAULT: default type of rules to test, * means all
+RULES_TYPE_DEFAULT="*"
```

Comment:
Should this not bé RULES_TYPES["prometheus"]|RULES_TYPES["loki"] instead? This will otherwise validité helm templates that are not rules right?

Reply:
Also maybe we want to make sure rules ends with either rules.yml or logs.yml? Otherwise the tooling will not run on files ending with rules.yaml or simple.yaml if people don't follow the convention right?

Reply:
This is the selector for the `RULES_TYPES` map, `*` select all items in the map so both prometheus and loki, they are later combined into the `rules_suffix_pattern` which lead to something like `.*(rules.yml|logs.yml)`

Reply:
Ah okay I did not check that thanks :)

Reply:
The PR looks overall fine to me but i'm no bash expert

Reply:
Enforcing naming convention would be good if possible, but let's have this as a separate PR.

Reply:
Let's wait for the wizard :mage:

Reply:
Yes sure but can you still do that later?

Reply:
Also don't forget to update the readme on how to test rules

Reply:
> Yes sure but can you still do that later?

I added a todo in the issue

> Also don't forget to update the readme on how to test rules

There's already a new section in the readme, do you expect something else ?

Reply:
Wait you did :facepalm: